app/models/QualifierOfIdentification.scala

```diff
@@ -56,6 +58,13 @@ object QualifierOfIdentification extends EnumerableType[QualifierOfIdentificatio
 
   val messageKeyPrefix: String = "qualifierOfIdentification"
 
+  def values(userAnswers: UserAnswers): Seq[QualifierOfIdentification] =
+    userAnswers.get(TypeOfLocationPage) match {
+      case Some(DesignatedLocation) => Seq(CustomsOffice, Unlocode)
+      case Some(ApprovedPlace)      => Seq(PostalCode, Unlocode, Coordinates, EoriNumber, Address)
+      case Some(Other)              => Seq(PostalCode, Unlocode, Coordinates, Address)
```

Comment:
You may want a catch all case here to avoid a matcher error (im sure this wont ever happen but better safe than sorry)

Reply:
Oh crikey yes I didn't spot that ^ Could just return all values for that case